piano-hero/mobile/src/main/java/com/novoda/pianohero/GameModel.java

```diff
@@ -0,0 +1,101 @@
+package com.novoda.pianohero;
+
+import android.support.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class GameModel implements GameMvp.Model {
+
+    private final SongSequenceFactory songSequenceFactory;
+    private final SimplePitchNotationFormatter pitchNotationFormatter;
+
+    private Sequence sequence;
+
+    GameModel(
+        SongSequenceFactory songSequenceFactory,
+        SimplePitchNotationFormatter pitchNotationFormatter) {
+        this.songSequenceFactory = songSequenceFactory;
+        this.pitchNotationFormatter = pitchNotationFormatter;
+    }
+
+    @Override
+    public void startGame(StartCallback callback) {
+        sequence = songSequenceFactory.maryHadALittleLamb();
+
+        checkSequenceIsSimpleElseThrow(sequence);
+
+        callback.onGameStarted(createViewModel(sequence));
+    }
+
+    private void checkSequenceIsSimpleElseThrow(Sequence sequence) {
+        for (int i = 0; i < sequence.length(); i++) {
+            Notes notes = sequence.get(i);
+            if (notes.count() > 1) {
+                throw new IllegalArgumentException("Sequence contains chords, that's not simple enough");
+            }
+            for (Note note : notes.notes()) {
+                if (pitchNotationFormatter.format(note).endsWith("#")) {
+                    throw new IllegalArgumentException("Sequence contains sharps, that's not simple enough");
+                }
+            }
+        }
+    }
+
+    @NonNull
+    private RoundViewModel createViewModel(Sequence updatedSequence) {
```

Comment:
the Types are erased at run time :trollface: 

`converter.convert(domainModel)` has to happen somewhere, just have to deal with it that its here (as this will be off the mainthread SOON) any higher up will be MT